### PR TITLE
Use 64 bit wasm_runtime_enlarge_memory() increment

### PR DESCRIPTION
**Pull request by bnason-nf**
_Originally posted by @yamt in https://github.com/bytecodealliance/wasm-micro-runtime/pull/3569#discussion_r1654398315_

### CODE REVIEW

File: core/iwasm/common/wasm_memory.c [hunk 1]
```diff
@@ -921,7 +921,7 @@ wasm_enlarge_memory_internal(WASMModuleInstance *module, uint32 inc_page_count)
 
 bool
 wasm_runtime_enlarge_memory(WASMModuleInstanceCommon *module_inst,
-                            uint32_t inc_page_count)
+                            uint64 inc_page_count)
 {
```
Comment (yamt) [inline core/iwasm/common/wasm_memory.c:925]: please reject inc_page_count > UINT32_MAX rather than truncating silently.

Comment (bnason-nf) [reply core/iwasm/common/wasm_memory.c:925]: @yamt: I have done this, but it doesn't protect against `cur_page_count + inc_page_count > UINT32_MAX`, so is really not protecting much at all, and will mean that the person who works on true 64 bit support will need to remove this check.

Comment (yamt) [reply core/iwasm/common/wasm_memory.c:925]: > but it doesn't protect against `cur_page_count + inc_page_count > UINT32_MAX`,

really?

i thought this logic was about it. https://github.com/bytecodealliance/wasm-micro-runtime/blob/65bf04e98643f9cbccd6cdc5b8e6d82029c52c38/core/iwasm/common/wasm_memory.c#L789-L792

Comment (bnason-nf) [reply core/iwasm/common/wasm_memory.c:925]: @yamt: oh good, I missed that. So then the overflow isn't an issue, just the need to remove this check in the future.

**Review by yamt — APPROVED** [review]
lgtm

**Pull request merged**